src/core/Directus/Database/Schema/Object/FieldRelationship.php

```diff
@@ -127,7 +139,7 @@ protected function guessType()
             $this->getFieldOne()        === $fieldName &&
             $this->getCollectionOne()   !== null
         ) {
-            $type = static::ONE_TO_MANY;
+            $type = ($interface === 'many-to-many') ? static::MANY_TO_MANY : static::ONE_TO_MANY;
```

Comment:
To reiterate Welling's previous statement, while this would work just fine, it would lock the functionality down to the interface called `many-to-many`. There will most likely be more many to many style interfaces with different input output formats (think tags, or a specialized "favorites" interface)